original_post_discovery.py

```diff
@@ -117,10 +117,10 @@ def resolve(urls):
     # the best chance of finding a match. Some silos allow several
     # different permalink formats to point to the same place (e.g.,
     # facebook user id instead of user name)
-    syndication_url = source.canonicalize_syndication_url(
-      util.follow_redirects(syndication_url).url)
-    originals.update(_posse_post_discovery(
-      source, activity, syndication_url, fetch_hfeed))
+    syndication_url = source.canonicalize_url(syndication_url)
+    if syndication_url:
+      originals.update(_posse_post_discovery(
+        source, activity, syndication_url, fetch_hfeed))
```

Comment:
oh no, it still follows redirects if the URL isn't known canonical. i just moved that into the new UrlCanonicalizer.